src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -229,6 +229,34 @@ class E
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(49628, "https://github.com/dotnet/roslyn/issues/49628")]
+        public void AmbigCtor()
+        {
+            var src = @"
+record R(R x);
+
+#nullable enable
+record R2(R2? x) { }
```

Comment:
What happens if this parameter is referenced in a field initializer. What information is going to be returned for that reference by SemanticModel?